src/Native/gc/gc.cpp

```diff
@@ -17833,12 +17833,12 @@ uint8_t* gc_heap::find_object (uint8_t* interior)
     {
         // this is a pointer to a UOH object
         heap_segment* seg = find_segment (interior, FALSE);
-        if (seg
+        if (seg)
+        {
 #ifdef FEATURE_CONSERVATIVE_GC
-            && (GCConfig::GetConservativeGC() || interior <= heap_segment_allocated(seg))
+            if ( interior >= heap_segment_allocated(seg))
```

Comment:
This is the same as merged in https://github.com/dotnet/runtime/pull/39905 .  If this complicates merging, I can leave this PR as WIP.